fastlane_core/lib/fastlane_core/itunes_transporter.rb

```diff
@@ -227,4 +158,151 @@ def shell_escaped_password(password)
+      [
+        Helper.transporter_java_executable_path.shellescape,
+        "-Djava.ext.dirs=#{Helper.transporter_java_ext_dir.shellescape}",
+        '-XX:NewSize=2m',
+        '-Xms32m',
+        '-Xmx1024m',
+        '-Xms1024m',
+        '-Djava.awt.headless=true',
+        '-Dsun.net.http.retryPost=false',
+        "-classpath #{Helper.transporter_java_jar_path.shellescape}",
+        'com.apple.transporter.Application',
+        '-m lookupMetadata',
+        "-u #{username.shellescape}",
+        "-p #{password.shellescape}",
+        "-apple_id #{apple_id.shellescape}",
+        "-destination #{destination.shellescape}",
+        '2>&1' # cause stderr to be written to stdout
+      ].join(' ')
+    end
+
+    def execute(command, hide_output)
+      # The Java command needs to be run starting in a working directory in the iTMSTransporter
+      # file area. The shell script takes care of changing directories over to there, but we'll
+      # handle it manually here for this strategy.
+      FileUtils.cd(Helper.itms_path) do
+        return super(command, hide_output)
+      end
+    end
+  end
+
+  class ItunesTransporter
+    # This will be called from the Deliverfile, and disables the logging of the transporter output
+    def self.hide_transporter_output
+      @hide_transporter_output = !$verbose
+    end
+
+    def self.hide_transporter_output?
+      @hide_transporter_output
+    end
+
+    # Returns a new instance of the iTunesTransporter.
+    # If no username or password given, it will be taken from
+    # the #{CredentialsManager::AccountManager}
+    def initialize(user = nil, password = nil, avoid_shell_script = false)
+      data = CredentialsManager::AccountManager.new(user: user, password: password)
+      @user = data.user
+      @password = data.password
+      @transporter_executor = avoid_shell_script ? JavaTransporterExecutor.new : ShellScriptTransporterExecutor.new
+    end
+
+    # Downloads the latest version of the app metadata package from iTC.
+    # @param app_id [Integer] The unique App ID
+    # @param dir [String] the path in which the package file should be stored
+    # @return (Bool) True if everything worked fine
+    # @raise [Deliver::TransporterTransferError] when something went wrong
+    #   when transfering
+    def download(app_id, dir = nil)
+      dir ||= "/tmp"
+
+      UI.message("Going to download app metadata from iTunes Connect")
+      command = @transporter_executor.build_download_command(@user, @password, app_id, dir)
+      UI.verbose(@transporter_executor.build_download_command(@user, 'YourPassword', app_id, dir)) if $verbose
+
+      result = @transporter_executor.execute(command, ItunesTransporter.hide_transporter_output?)
+
+      itmsp_path = File.join(dir, "#{app_id}.itmsp")
+      if result and File.directory? itmsp_path
+        UI.success("Successfully downloaded the latest package from iTunes Connect.")
```

Comment:
so if `result` is true, and the directory isn't created, we still return `true` but display an error ?